src/component/ResponsiveContainer.tsx

```diff
@@ -60,28 +67,31 @@ export const ResponsiveContainer = forwardRef(
         containerWidth: containerRef.current.clientWidth,
         containerHeight: containerRef.current.clientHeight,
       };
-    };
-
-    const updateDimensionsImmediate = () => {
-      if (!mounted) {
-        return;
-      }
+    }, []);
 
+    const updateDimensionsImmediate = useCallback(() => {
       const newSize = getContainerSize();
 
       if (newSize) {
-        const { containerWidth: oldWidth, containerHeight: oldHeight } = sizes;
         const { containerWidth, containerHeight } = newSize;
 
-        if (containerWidth !== oldWidth || containerHeight !== oldHeight) {
-          setSizes({ containerWidth, containerHeight });
-        }
+        setSizes(currentSizes => {
+          const { containerWidth: oldWidth, containerHeight: oldHeight } = currentSizes;
+          if (containerWidth !== oldWidth || containerHeight !== oldHeight) {
+            return { containerWidth, containerHeight };
+          }
+
+          return currentSizes;
+        });
       }
-    };
+    }, [getContainerSize]);
 
-    const handleResize = debounce > 0 ? _.debounce(updateDimensionsImmediate, debounce) : updateDimensionsImmediate;
+    const handleResize = useMemo(
+      () => (debounce > 0 ? _.debounce(updateDimensionsImmediate, debounce) : updateDimensionsImmediate),
+      [debounce, updateDimensionsImmediate],
+    );
 
-    const renderChart = () => {
+    const computedSizes = useMemo(() => {
```

Comment:
Would it make sense to extract this as a unit testable helper function, now that it only computes the size?

Reply:
I don't think this is necessary, right now the majority of tests of `ResponsiveContainer` involve this function.
We might consider to move it away if we have to use it on a different component, but for the moment I think is fine to leave it here.

Reply:
I think its fine to leave here for now as well